lang/funcs/core/sys/uptime_fact_darwin.go

```diff
@@ -0,0 +1,42 @@
+// Mgmt
+// Copyright (C) 2013-2019+ James Shubin and the project contributors
+// Written by James Shubin <james@shubin.ca> and the project contributors
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+// +build darwin
+
+package coresys
+
+import (
+	"fmt"
+	"syscall"
+	"time"
+	"unsafe"
+
+	"golang.org/x/sys/unix"
+)
+
+func uptime() (int64, error) {
+	out, err := unix.SysctlRaw("kern.boottime")
+	if err != nil {
+		return 0, err
+	}
+	var boottime syscall.Timeval
+	if len(out) != int(unsafe.Sizeof(boottime)) {
+		return 0, fmt.Errorf("Failed to get boottime")
```

Comment:
error messages should not be capitalized. and we should say "uptime" not boottime.

Reply:
well technically its the time-at-boot we failed to fetch. It's only later that it's converted into the uptime by subtracting it from the current time since epoch.